### PR TITLE
update applications post and put apis, add tests

### PR DESCRIPTION
**Pull request by AlexAzalor**

### CODE REVIEW

File: api/controllers/application.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            m.Application.status == m.ApplicationStatus.PENDING,
+            m.Application.id != application.id,
+        )
+    ).all()
+
+    if job_applications:
+        for job_app in job_applications:
+            job_app.status = m.ApplicationStatus.REJECTED
+
+        log(log.INFO, "Rejected applications count: [%s]", len(job_applications))
```
Comment (Taraslysak) [inline api/controllers/application.py]: wrong indentation. This will trigger for every rejected application

File: api/controllers/application.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import app.models as m
+from config import config
+from app.logger import log
+
+
+CFG = config()
+
+
+def reject_applications(db: Session, application: m.Application):
```
Comment (Taraslysak) [inline api/controllers/application.py]: rename to `reject_other_not_accepted_applications`

File: api/routes/application.py [hunk 3]
```diff
@@ -18,23 +19,25 @@ def get_application(
...
+    )
     if not application:
         log(log.ERROR, "Application [%s] not found", application_id)
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Application not found")
+
     return application
 
 
-@application_router.get("/", status_code=status.HTTP_200_OK, response_model=s.ApplicationOutList)
+@application_router.get("/list/{job_id}", status_code=status.HTTP_200_OK, response_model=s.ApplicationOutList)
```
Comment (Taraslysak) [inline api/routes/application.py]: I think job_id can be query argument, not path parameter

File: api/routes/application.py [hunk 4]
```diff
@@ -77,37 +85,94 @@ def create_application(
...
+)
 def update_application(
     application_id: int,
-    data: s.ApplicationPut,
+    data: s.ApplicationPutIn,
     db: Session = Depends(get_db),
     current_user: m.User = Depends(get_current_user),
 ):
-    application: m.Application | None = db.scalar(sa.select(m.Application).where(m.Application.id == application_id))
+    if not data.status:
```
Comment (Taraslysak) [inline api/routes/application.py]: This check is redundant. Just make sure that status is not optional field in `ApplicationPutIn` Pydantic will do the validation for you

File: api/routes/application.py [hunk 5]
```diff
@@ -77,37 +85,94 @@ def create_application(
...
         log(log.ERROR, "Application [%s] not found", application_id)
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Application not found")
-    job_owner_id: int | None = db.scalar(sa.select(m.Job.owner_id).where(m.Job.id == application.job_id))
-    if not job_owner_id:
+
+    if application.status in [m.ApplicationStatus.ACCEPTED, m.ApplicationStatus.REJECTED]:
+        log(log.ERROR, "Application [%s] is already completed", application_id)
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Application is already completed")
+
+    job = db.scalar(sa.select(m.Job).where(m.Job.id == application.job_id))
```
Comment (Taraslysak) [inline api/routes/application.py:114]: application should have relationship on job. No need to do extra DB query here, just do `application.job...`

File: app/schema/application.py [hunk 6]
```diff
@@ -39,9 +39,21 @@ class ApplicationIn(BaseModel):
...
     model_config = ConfigDict(
         from_attributes=True,
     )
+
+
+class ApplicationPutOut(BaseModel):
+    application: ApplicationOut
+
+
+class ApplicationListIn(BaseModel):
```
Comment (Taraslysak) [inline app/schema/application.py]: Not sure we need this class

**Review by Taraslysak — COMMENTED** [review]
Please consider somer refactoring

File: api/routes/application.py (hunk 5 above)
Comment (AlexAzalor) [reply api/routes/application.py:114]: application hasn't any relationships

**Review by Taraslysak — APPROVED** [review]

**Pull request merged**